OakCityV2/TableViewController.m

```diff
@@ -0,0 +1,216 @@
+//
+//  TableViewController.m
+//  OakCityV2
+//
+//  Created by Josh Green on 6/8/15.
+//  Copyright (c) 2015 Josh Green. All rights reserved.
+//
+
+#import "TableViewController.h"
+#import "AFNetworking/AFNetworking.h"
+#import "DetailViewController.h"
+#import "DetailCell.h"
+#import "Job.h"
+#import "AFNetworking/UIImageView+AFNetworking.h"
+#define LANGUAGE @"PHP"
+#define TOWN @"San+Francisco"
+
+@interface TableViewController ()
+
+@end
+
+@implementation TableViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    //Fetch JSON
+    NSString *urlAsString = [NSString stringWithFormat:@"https://jobs.github.com/positions.json?description=%@&location=%@", LANGUAGE, TOWN];
+    NSURL *url = [NSURL URLWithString:urlAsString];
+    NSURLRequest *request = [NSURLRequest requestWithURL: url];
+    AFHTTPRequestOperation *operation = [[AFHTTPRequestOperation alloc] initWithRequest:request];
+    operation.responseSerializer = [AFJSONResponseSerializer serializer];
+    
+    //Parse JSON
+    [operation setCompletionBlockWithSuccess:^(AFHTTPRequestOperation *operation, id responseObject) {
+        self.jobs  = [NSMutableArray new];
+        NSArray *responseArray = (NSArray *)responseObject;
+        
+        for (NSDictionary *responseDict in responseArray) {
+        Job* job = [[Job alloc] initWithDictionary:responseDict];
+        [self.jobs addObject:job];
+        }
+        
```

Comment:
After you've finished building the list, you'll set self.jobs like

self.jobs = [NSArray arrayWithArray:tmpArray];

This makes a new (immutable) NSArray with the contents of the mutable tmpArray.
Technically, you could just do 

self.jobs = tmpArray;

Since NSMutableArray (tmpArray) is a subclass of NSArray (self.jobs) this is perfectly legal and the compiler will error if you try to use self.jobs as a mutableArray anywhere else.

BUT!  Sometimes you'll see code that actually checks to see what class an object in and act accordingly.  So it's safer just to make an immutable array to store in self.jobs.